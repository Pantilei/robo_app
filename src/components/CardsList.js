import React from "react";
import Card from "./Card";

const CardsList = ({ robots }) => {
  const robotsList = robots.map(item => {
    return (
      <Card key={item.id} name={item.name} email={item.email} id={item.id} />
    );
  });
  /* if (true) {
    throw new Error("Noooooooo............!");
  } */
  return <div>{robotsList}</div>;
};

export default CardsList;
