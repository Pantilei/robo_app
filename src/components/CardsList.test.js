import React from "react";
import { shallow } from "enzyme";
import CardsList from "./CardsList";

it("Testing CardsList component", () => {
  const mockRobots = [{ id: 1, name: "John Doe", email: "john@gmail.com" }];

  expect(shallow(<CardsList robots={mockRobots} />)).toMatchSnapshot();
});
