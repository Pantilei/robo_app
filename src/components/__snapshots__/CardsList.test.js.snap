// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CardsList component 1`] = `ShallowWrapper {}`;
